samples/matter/common/src/bridge/bridge_manager.cpp

```diff
@@ -165,48 +129,48 @@ CHIP_ERROR BridgeManager::AddDevice(BridgedDevice *device)
 	return CHIP_ERROR_NO_MEMORY;
 }
 
-BridgedDevice *BridgeManager::GetBridgedDevice(uint16_t endpoint, const EmberAfAttributeMetadata *attributeMetadata,
-					       uint8_t *buffer)
+BridgeManager::DevicePtr *
+BridgeManager::GetBridgedDevice(uint16_t index, const EmberAfAttributeMetadata *attributeMetadata, uint8_t *buffer)
 {
-	if (endpoint >= kMaxBridgedDevices) {
+	if (index >= kMaxBridgedDevices) {
 		return nullptr;
 	}
 
-	BridgedDevice *device = sBridgeManager.mBridgedDevices[endpoint];
-	if ((device == nullptr) || !buffer || !attributeMetadata) {
+	DevicePtr *device = const_cast<DevicePtr *>(&(sBridgeManager.mDevicesMap[index]->mDevice));
+	if (!device || !buffer || !attributeMetadata) {
 		return nullptr;
 	}
 	return device;
 }
 
-CHIP_ERROR BridgeManager::HandleRead(uint16_t endpoint, ClusterId clusterId,
+CHIP_ERROR BridgeManager::HandleRead(uint16_t index, ClusterId clusterId,
 				     const EmberAfAttributeMetadata *attributeMetadata, uint8_t *buffer,
 				     uint16_t maxReadLength)
 {
-	BridgedDevice *device = GetBridgedDevice(endpoint, attributeMetadata, buffer);
+	DevicePtr *device = GetBridgedDevice(index, attributeMetadata, buffer);
 
 	if (!device) {
 		return CHIP_ERROR_INVALID_ARGUMENT;
 	}
 
-	return device->HandleRead(clusterId, attributeMetadata->attributeId, buffer, maxReadLength);
+	return (*device)->HandleRead(clusterId, attributeMetadata->attributeId, buffer, maxReadLength);
```

Comment:
Should `GetBridgedDevice` return `BridgedDevice *` to avoid double pointer dereference?

Reply:
I'll just remove the `GetBridgedDevice` for the sake of direct access to the map with []operator.